File: packages/android-kotlin/src/main/kotlin/com/cinacoin/appkit/ui/CinacoinAccountCard.kt
package com.cinacoin.appkit.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Logout
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cinacoin.appkit.config.ChainRegistry
import com.cinacoin.appkit.config.CinacoinTheme

/**
 * Card displaying connected wallet account info
 */
@Composable
fun CinacoinAccountCard(
    address: String,
    chainId: Int? = null,
    onDisconnect: () -> Unit = {},
    theme: CinacoinTheme = if (androidx.compose.foundation.isSystemInDarkTheme()) 
        CinacoinTheme.Dark else CinacoinTheme.Light
) {
    val chain = chainId?.let { ChainRegistry.getChain(it) }
    
    Surface(
        modifier = Modifier
            .fillMaxWidth()
            .border(1.dp, theme.border, RoundedCornerShape(12.dp)),
        shape = RoundedCornerShape(12.dp),
        color = theme.surface
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = shortenAddress(address),
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    fontFamily = FontFamily.Monospace,
                    color = theme.text
                )
                if (chain != null) {
                    Spacer(modifier = Modifier.height(4.dp))
                    Text(
                        text = chain.name,
                        fontSize = 12.sp,
                        color = theme.textSecondary
                    )
                }
            }
            
            IconButton(onClick = onDisconnect) {
                Icon(
                    imageVector = Icons.Default.Logout,
                    contentDescription = "Disconnect",
                    tint = theme.textSecondary
                )
            }
        }
    }
}

private fun shortenAddress(address: String): String {
    if (address.length <= 10) return address
    return "${address.take(6)}...${address.takeLast(4)}"
}

File: packages/flutter-dart/lib/appkit_ui/cinacoin_account_card.dart
import 'package:flutter/material.dart';
import '../appkit_config/chain_registry.dart';
import '../appkit_config/cinacoin_theme.dart';

/// Card displaying connected wallet account info
class CinacoinAccountCard extends StatelessWidget {
  final String address;
  final int? chainId;
  final VoidCallback? onDisconnect;
  final CinacoinTheme? theme;

  const CinacoinAccountCard({
    super.key,
    required this.address,
    this.chainId,
    this.onDisconnect,
    this.theme,
  });

  @override
  Widget build(BuildContext context) {
    final themeData = theme ?? _getThemeFromBrightness(context);
    final chain = chainId != null ? ChainRegistry.getChain(chainId!) : null;

    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: themeData.surface,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: themeData.border),
      ),
      child: Row(
        children: [
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(
                  _shortenAddress(address),
                  style: TextStyle(
                    fontSize: 16,
                    fontWeight: FontWeight.w600,
                    fontFamily: 'monospace',
                    color: themeData.text,
                  ),
                ),
                if (chain != null) ...[
                  const SizedBox(height: 4),
                  Text(
                    chain.name,
                    style: TextStyle(
                      fontSize: 12,
                      color: themeData.textSecondary,
                    ),
                  ),
                ],
              ],
            ),
          ),
          if (onDisconnect != null)
            IconButton(
              onPressed: onDisconnect,
              icon: Icon(Icons.logout, color: themeData.textSecondary),
            ),
        ],
      ),
    );
  }

  String _shortenAddress(String address) {
    if (address.length <= 10) return address;
    return '${address.substring(0, 6)}...${address.substring(address.length - 4)}';
  }

  CinacoinTheme _getThemeFromBrightness(BuildContext context) {
    final brightness = Theme.of(context).brightness;
    return brightness == Brightness.dark 
        ? CinacoinTheme.dark 
        : CinacoinTheme.light;
  }
}
